.github/workflows/ci.yml

```diff
@@ -0,0 +1,575 @@
+        env:
+          # Fix linking of `linkme`: https://github.com/dtolnay/linkme/issues/49
+          RUSTFLAGS:                     -Clink-arg=-z -Clink-arg=nostart-stop-gc
+        run: |
+          # run all tests with --all-features, which will run the e2e-tests feature if present
+          scripts/for_all_contracts_exec.sh --path integration-tests --ignore static-buffer -- cargo test \
+            --verbose --all-features --manifest-path {}
+          # run the static buffer test with a custom buffer size
+          cargo clean --manifest-path integration-tests/static-buffer/Cargo.toml
+          INK_STATIC_BUFFER_SIZE=30 cargo test --verbose --manifest-path integration-tests/static-buffer/Cargo.toml --all-features
+
+  examples-contract-build:
+    runs-on: ubuntu-latest
+    needs: [build]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
+    strategy:
+      fail-fast: false
+      matrix:
+        type: [WASM, RISCV]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-directories: ${{ env.CARGO_TARGET_DIR }}
+
+      - name: Rust Info
+        uses: ./.github/rust-info
+
+      - name: Build Contract WASM Examples
+        if: ${{ matrix.type == 'WASM' }}
+        run: |
+          rustup component add rust-src --toolchain stable
+          cargo contract -V
+          # Build all examples
+          scripts/for_all_contracts_exec.sh --path integration-tests -- cargo contract build --release --manifest-path {}
+          # Build the different features for the conditional compilation example
+          pushd ./integration-tests/conditional-compilation
+          cargo contract build --release --features "foo"
+          cargo contract build --release --features "bar"
+          cargo contract build --release --features "foo, bar"
+          popd
+
+      - name: Build Contract RISCV Examples
+        if: ${{ matrix.type == 'RISCV' }}
+        env:
+          RUSTC_BOOTSTRAP:               1
+        run: |
+          rustup component add rust-src --toolchain stable
+          cargo contract -V
+          # We skip some examples for those reasons:
+          # This uses dlmalloc which is only supported on select targets.
+          #   - custom_allocator
+          # Pulls in sp-std which needlessly requires atomic pointers (TODO: Fix sp-std and enable this example)
+          #   - call-runtime
+          scripts/for_all_contracts_exec.sh --path integration-tests --ignore custom-allocator --ignore call-runtime \
+            -- cargo build --manifest-path {} --no-default-features --target $RISCV_TARGET -Zbuild-std="core,alloc"
+
+  examples-docs:
+    runs-on: ubuntu-latest
+    needs: [build]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
```

Comment:
```suggestion
      image: ${{ env.IMAGE }}
```

Reply:
I think that in this context, environment variables are not being handled